src/quo2/components/avatars/user_avatar.cljs

```diff
@@ -84,71 +76,70 @@
    :height        inner-dimensions
    :border-radius inner-dimensions})
 
+(defn outer-styles
+  [outer-dimensions]
+  {:width         outer-dimensions
+   :height        outer-dimensions
+   :border-radius outer-dimensions})
+
 (defn container
-  [inner-dimensions outer-dimensions & children]
-  [rn/view
-   {:style (merge {:background-color (colors/custom-color-by-theme :turquoise 50 60)
-                   :justify-content  :center
-                   :align-items      :center}
-                  (container-styling inner-dimensions outer-dimensions))}
-   children])
+  [{:keys [inner-dimensions outer-dimensions]} & children]
+  (-> [rn/view
+       {:style (merge
+                {:background-color (colors/custom-color-by-theme :turquoise 50 60)
+                 :justify-content  :center
+                 :align-items      :center}
+                (container-styling inner-dimensions outer-dimensions))}]
+      (concat children)
```

Comment:
Is this function ever going to receive anything other than the single `[text/text ...` below? If so, I'd look at simplifying this, maybe just including the code in the `initials-avatar` function that seems to be the only thing that uses it.

Reply:
I usually follow the pattern @ulisesmac used here when I want the function to show it doesn't care what/who are the children, if it's just a text component, etc.

Going a bit on a tangent, this new library https://github.com/pitch-io/uix tries to help developers write components in a more uniform way https://github.com/pitch-io/uix/blob/master/docs/components.md. Reagent, on the other hand, is much more open. Different approaches.

Reply:
@erikseppanen  That component is not used in any other part of the code. In the previous implementation, it was used twice in this namespace. I wanted to keep general existing code structure and not to perform a huge refactor over the file when fixing the current issue, but I agree with you.

@ilmotta  Yes, sometimes it's better than just using children as is, (e.g. in the previous implementation the child text required the `^key` even though it was a single child).

Anyway, in 9b0176641bced3ce44395de303a4fdaad381c63a I removed the `container` component and renamed the `container-styling` function since both were used just once, the code is now smaller and IMO clearer.

Reply:
@ilmotta Have you tried that library? if so, what do you think?

I've just checked it, it looks as using hiccup but replacing `[comp props]` by `($ comp props)`.
> tries to help developers write components in a more uniform way

Have you found an example you could share to us?

thanks!

Reply:
> in the previous implementation the child text required the ^key even though it was a single child

:+1: If we use the pattern `(into [comp-a props] children)` then React won't complain about missing unique keys.

> I've just checked it, it looks as using hiccup but replacing [comp props] by ($ comp props)

I have zero experience with that lib, but I'm interested, especially by the performance claims, but also because of some design choices. I didn't know it used hiccup behind the scenes. Apparently, the performance improvements are in great part achieved by the macro `$`, which can ahead of time compile and optimize a bunch of stuff. Reagent also has hiccup a compiler, but it's slower. Very interesting! Always useful to get outside the Reagent bubble. https://github.com/pitch-io/uix/blob/25cc16a5787d30492e17b7d3e254d351e45fc6e9/core/src/uix/core.clj#L128

>tries to help developers write components in a more uniform way

By uniformity I mean the `defui` macro is telling developers there's one specific way to write a component. With Reagent it's much more open for interpretation. In React, every component that needs arguments will receive them via a map. The ability to use `children` is also implicitly supported. In Reagent every component can be called in any way, and to use children we need to do the dance of wrapping in a `into` or `concat` call. Also in Reagent we have some weird stuff like using `reagent/current-component` and `reagent/props` to extract data from the component that's not apparent from the component/function signature. It all feels like workarounds...

So based on my experience with React, I see the API uniformity of uix as an advantage, as well as other libraries out there. Maybe I'm being too picky ;) 

Thanks for asking. I'd be interested in further discussions around these topics!